src/api/app/components/watchlist_component.rb

```diff
@@ -14,8 +14,11 @@ class WatchlistComponent < ApplicationComponent
   def initialize(user:, project: nil, package: nil, bs_request: nil)
     super
 
+    # NOTE: package can be also a String. For example, the live build log for multibuild packages.
+    package = Package.find_by_project_and_name(project.name, Package.striping_multibuild_suffix(package)) if package.is_a?(String)
```

Comment:
If we provide a package which isn't really a package, but a string to this component, we should change the code where this component is instantiated.

A view component should deal with view code, not fetching data from the database. We already do this here in methods like `projects`, `packages` and `bs_requests`, but this should be refactored too in another PR.

Reply:
Good advice @dmarcoux, thanks! I created a [card in the backlog](https://trello.com/c/toTBZ8aS/2529-extract-retrieving-objects-from-db-from-view-component) for this.

Reply:
Thank you @saraycp :slightly_smiling_face:

Reply:
Finally I opted for revisiting the package controller methods related to the live build log, and make sure that:
- `@package` always returns an object.
- the new instance variable `@package_name` keeps the package name to be shown in the views. It will contain what is passed from `param[:package]`.

For example, having `params[:package] = 'obs-server:flavor1'`:
- `@package` will contain the package object 'obs-server'. There isn't any package object for 'obs-server:flavor1'.
- `@package_name` will be the string 'obs-server:flavor1'.

I also renamed the `package` variables in `mixins/build_log_support.rb` to `package_name`, to make it clear that they are strings.